zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/usermanagement/AuthorizationQueryControllerTest.java

```diff
@@ -0,0 +1,280 @@
+    // when / then
+
+    webClient
+        .post()
+        .uri(AUTHORIZATION_SEARCH_URL)
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .json(EXPECTED_SEARCH_RESPONSE);
+
+    verify(authorizationServices).search(new AuthorizationQuery.Builder().build());
+  }
+
+  @Test
+  void shouldSearchAuthorizationsWithEmptyQuery() {
+    // given
+    when(authorizationServices.search(any(AuthorizationQuery.class)))
+        .thenReturn(SEARCH_QUERY_RESULT);
+    final String request = "{}";
+    // when / then
+    webClient
+        .post()
+        .uri(AUTHORIZATION_SEARCH_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .json(EXPECTED_SEARCH_RESPONSE);
+
+    verify(authorizationServices).search(new AuthorizationQuery.Builder().build());
+  }
+
+  @Test
+  void shouldSearchAuthorizationsWithSorting() {
+    // given
+    when(authorizationServices.search(any(AuthorizationQuery.class)))
+        .thenReturn(SEARCH_QUERY_RESULT);
+    final var request =
+        """
+            {
+                "sort": [
+                    {
+                        "field": "ownerType",
+                        "order": "desc"
+                    }
+                ]
+            }""";
+    // when / then
+    webClient
+        .post()
+        .uri(AUTHORIZATION_SEARCH_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isOk()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_JSON)
+        .expectBody()
+        .json(EXPECTED_SEARCH_RESPONSE);
+
+    verify(authorizationServices)
+        .search(
+            new AuthorizationQuery.Builder()
+                .sort(new AuthorizationSort.Builder().ownerType().desc().build())
+                .build());
+  }
+
+  @ParameterizedTest
+  @MethodSource("invalidAuthorizationSearchQueries")
+  void shouldInvalidateAuthorizationsSearchQueryWithBadQueries(
+      final String request, final String expectedResponse) {
+    // when / then
+    webClient
+        .post()
+        .uri(AUTHORIZATION_SEARCH_URL)
+        .accept(MediaType.APPLICATION_JSON)
+        .contentType(MediaType.APPLICATION_JSON)
+        .bodyValue(request)
+        .exchange()
+        .expectStatus()
+        .isBadRequest()
+        .expectHeader()
+        .contentType(MediaType.APPLICATION_PROBLEM_JSON)
+        .expectBody()
+        .json(expectedResponse);
+
+    verify(authorizationServices, never()).search(any(AuthorizationQuery.class));
+  }
+
+  public static Stream<Arguments> invalidAuthorizationSearchQueries() {
```

Comment:
👍 This is really nice. You get a direct overview of bad queries we test and what the result will be, love it 🦖